src/main/io/serial_cli.c

```diff
@@ -2800,6 +2833,34 @@ static void printConfig(char *cmdline, bool doDiff)
     }
 
     createDefaultConfig(&defaultConfig);
+    // make copies of configs to do differencing
```

Comment:
Yes, I think there are some possibilities to improve this. I'm adopting the "get it working then get it working well" approach here.